internal/nostr/nostr.go

```diff
@@ -126,11 +115,23 @@ func NIP47Handler(c echo.Context) error {
 	}
 }
 
-func processRequest(ctx context.Context, requestData *NIP47Request) (*nostr.Event, int, error) {
-	relay, err := nostr.RelayConnect(ctx, requestData.RelayURL)
+func (svc *NostrService) getRelayConnection(ctx context.Context, customRelayURL string) (*nostr.Relay, bool, error) {
+	if customRelayURL != "" && customRelayURL != svc.config.DefaultRelayURL {
+		logrus.Infof("connecting to custom relay: %s...", customRelayURL)
+		relay, err := nostr.RelayConnect(ctx, customRelayURL)
+		return relay, true, err // true means custom and the relay should be closed
+	}
+	return svc.defaultRelay, false, nil
```

Comment:
regarding reconnection, maybe it's possible to have some code here that makes sure the relay is connected? so if at some point the default relay connection drops it will be reconnected here?

Reply:
Done!